zephyr/wrapper.c

```diff
@@ -72,18 +72,33 @@ __section(".heap_mem") static uint8_t __aligned(64) heapmem[HEAPMEM_SIZE];
 
 #else
 
-#define HEAPMEM_SHARED_SIZE	(HEAP_SYSTEM_SIZE + HEAP_RUNTIME_SIZE + \
-				HEAP_RUNTIME_SHARED_SIZE + HEAP_SYSTEM_SHARED_SIZE)
+/* TODO: heap size should be set based on linker output */
+#if (CONFIG_HP_MEMORY_BANKS < 16)
+/* e.g. APL */
+#define	HEAP_SIZE	0x30000
+#elif (CONFIG_HP_MEMORY_BANKS < 30)
+/* e.g. JSL */
+#define	HEAP_SIZE	0x80000
+#elif (CONFIG_HP_MEMORY_BANKS < 45)
+/* e.g. TGL-H */
+#define	HEAP_SIZE	0x100000
```

Comment:
can you clarify how this is handled in rimage, your changes only deal with 'tgl' and there's no specific memory size for TGL-H.

Reply:
> can you clarify how this is handled in rimage, your changes only deal with 'tgl' and there's no specific memory size for TGL-H.

The memory size is configured in the Kconfig, see it here:
https://github.com/thesofproject/sof/blob/8fc5247289db0093bbb9de369c1dcb7e0d19a109/src/arch/xtensa/configs/tgph_defconfig#L8

For rimage, we need to specify image_size according to the memory size, see it here:
https://github.com/thesofproject/rimage/blob/9a26e4558094f19f3f7becd89eb9e8a9a9dd82b9/config/tgl-h.toml#L6

Reply:
but you didn't modify the tgl-h.toml file in this PR, did you? see https://github.com/thesofproject/sof/pull/4521/commits/1b0757e01555c01cc288a5182571a919620dde1c

Reply:
> but you didn't modify the tgl-h.toml file in this PR, did you? see [1b0757e](https://github.com/thesofproject/sof/commit/1b0757e01555c01cc288a5182571a919620dde1c)

That's because the tgl-h.toml is already updated before my PR, the rimage is already updated to 9a26e4558094 in the sof upstream repo:

9a26e4558094 config: apl: change image_size to the real SRAM size
8a2ea0044dcf config: cnl: change image_size to the real SRAM size
1de9ccca3388 config: jsl: change image_size to the real SRAM size
f52a078aadf1 config: icl: change image_size to the real SRAM size
8073ea3ef93c config: tgl-h: change image_size to the real SRAM size
9e50a02f1b0d config: tgl: change image_size to the real SRAM size